plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -330,6 +330,11 @@ public ConnectorPageSource createPageSource(
                 effectivePredicate,
                 table.getNameMappingJson().map(NameMappingParser::fromJson),
                 partitionKeys);
+        ReaderPageSource dataPageSource = readerPageSourceWithRowPositions.getReaderPageSource();
+
+        if (dataPageSource.get().isFinished()) {
+            return new EmptyPageSource();
```

Comment:
Why? add a comment

Reply:
Switched the approach here to just wrap the DeleteFilter reading in a Supplier. I think that reads better